packages/astro/src/assets/vite-plugin-assets.ts

```diff
@@ -29,6 +30,17 @@ export default function assets({
 		// Expose the components and different utilities from `astro:assets` and handle serving images from `/_image` in dev
 		{
 			name: 'astro:assets',
+			outputOptions(outputOptions) {
+				// Specifically split out chunk for asset services to prevent TLA deadlock
+				// https://github.com/rollup/rollup/issues/4708
+				extendManualChunks(outputOptions, {
+					before(id) {
+						if (id.includes('astro/dist/assets/services')) {
```

Comment:
Will this hardcoding cause an issue for custom services?

Reply:
Would custom services import things like from `astro/assets/utils`? Then maybe we need to go one directory up and make it `astro/dist/assets` instead 🤔

Reply:
Hmm, admittedly, it's fairly unlikely

Reply:
I think either ways it should still work fine as we're primarily grouping our Astro code into one chunk. The issue before was that `services/sharp.ts` uses `services/service.ts`. But both were in two different chunks. `services/sharp.ts` was in it's own chunk, but `services/service.ts` got grouped with the markdown page chunk. And so there was an import loop, and the TLA in the markdown page made it worst.

So manually handling where `services/service.ts` should be chunked fixed it here. I don't mind if we want to go back one directory down though.

Reply:
I'm honestly not sure of the impact this can have, so I think whichever works just fine for me as long as it works, ha.

If it turns out to be a problem somehow, we can change it back fairly easily it seems like

Reply:
Changed it back to `astro/dist/assets/services/` for now since it feels a liitle safer that way 👍